solutions/observability/apm/resource-attributes.md

```diff
@@ -36,13 +40,119 @@ processors:
+
+Only a subset of OpenTelemetry resource attributes are directly mapped to ECS fields. If an attribute doesn't have a predefined ECS mapping, it's stored under `labels.*`, with `.` (dots) replaced by `_` (underscores).
+
+The following table shows how OTel resource attributes are mapped to ECS fields and how they're stored if unmapped.
+
+| OpenTelemetry attribute | Mapped ECS field | If unmapped, stored as |
+|------------------------|------------------|------------------------|
+| `service.name` | `service.name` | - |
+| `service.version` | `service.version` | - |
+| `deployment.environment` | `service.environment` | - |
+| `cloud.provider` | `cloud.provider` | - |
+| `cloud.account.id` | `cloud.account.id` | - |
+| `otel.library.name` | ❌ Not mapped | `labels.otel_library_name` |
+| `custom.attribute.with.dots` | ❌ Not mapped | `labels.custom_attribute_with_dots` |
+
+For example, if an OpenTelemetry resource contains:
+
+```json
+{
+  "service.name": "user-service",
+  "deployment.environment": "production",
+  "otel.library.name": "my-lib",
+  "custom.attribute.with.dots": "value"
+}
+```
+
+Elastic APM stores the following:
+
+```json
+{
+  "service.name": "user-service",
+  "service.environment": "production",
+  "labels": {
+    "otel_library_name": "my-lib",
+    "custom_attribute_with_dots": "value"
+  }
+}
+```
+
+## APM transactions and spans
+
+Not all OpenTelemetry spans are mapped the same way:
+
+- Root spans (entry points) are mapped to APM transactions.
+- Child spans (internal operations, DB queries) are mapped to APM spans.
 
-::::
+| OpenTelemetry span kind | Mapped to APM | Example |
+|------------------------|---------------|---------|
+| `SERVER` | Transaction | Incoming HTTP request (`GET /users/{id}`) |
+| `CONSUMER` | Transaction | Message queue consumer event |
+| `CLIENT` | Span | Outgoing database query (`SELECT * FROM users`) |
+| `PRODUCER` | Span | Sending a message to a queue |
+| `INTERNAL` | Span | Internal function execution |
 
-% Stateful only after this?
+The following example shows OpenTelemetry spans:
+
+```json
+[
+  {
+    "traceId": "abcd1234",
+    "spanId": "root5678",
+    "parentId": null,
+    "name": "GET /users/{id}",
+    "kind": "SERVER"
+  },
+  {
+    "traceId": "abcd1234",
+    "spanId": "db1234",
+    "parentId": "root5678",
+    "name": "SELECT FROM users",
+    "kind": "CLIENT"
+  }
+]
+```
+
+The previous OTel spans are stored by Elastic APM as follows:
+
+```
+Transaction: GET /users/{id}
+ ├── Span: SELECT FROM users
+```
+
+## Conditional attribute translation
+
+Some OpenTelemetry attributes are conditionally converted based on their value type.
+
+The following table shows how OTel resource attributes are converted.
+
+| OpenTelemetry Resource attribute | Incoming value type | Converted value | APM field |
+|------------------------|---------------------|----------------|-----------|
+| `http.status_code` | `200` (Integer) | `"200"` (String) | `http.response.status_code` |
```

Comment:
I glanced over the code and it seems to disagree. There's no type conversion for http.response.status_code . https://github.com/elastic/apm-data/blob/1e73b0cbd06208a4239e502cd5d512e9d0c0264d/input/otlp/traces.go#L647